src/main/java/cms/gov/madie/measure/services/TestCaseLockService.java

```diff
@@ -0,0 +1,98 @@
+package cms.gov.madie.measure.services;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.stereotype.Service;
+
+import com.nimbusds.oauth2.sdk.util.CollectionUtils;
+
+import cms.gov.madie.measure.repositories.MeasureRepository;
+import cms.gov.madie.measure.repositories.TestCaseLockRepository;
+import cms.gov.madie.measure.resources.DuplicateKeyException;
+import gov.cms.madie.models.measure.Measure;
+import gov.cms.madie.models.measure.TestCase;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import cms.gov.madie.measure.dto.TestCaseLock;
+import cms.gov.madie.measure.exceptions.ResourceNotFoundException;
+import cms.gov.madie.measure.dto.LockInfo;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class TestCaseLockService {
+  private final MeasureRepository measureRepository;
+  private final TestCaseLockRepository testCaseLockRepository;
+
+  public synchronized LockInfo lockTestCase(String measureId, String testCaseId, String userName) {
+    validateMeasureAndTestCase(measureId, testCaseId);
+
+    LockInfo lockInfo = null;
+    Instant now = Instant.now();
+    Instant expiresAt = now.plus(Duration.ofMinutes(15)); // 15 minute lock
+
+    TestCaseLock lock =
+        TestCaseLock.builder()
+            .measureId(measureId)
+            .testCaseId(testCaseId)
+            .lockedBy(userName)
+            .lockedAt(Instant.now())
+            .expiresAt(expiresAt)
+            .build();
+
+    try {
+      testCaseLockRepository.insert(lock);
+      lockInfo = LockInfo.builder().lockedId(testCaseId).isLocked(true).lockedBy(userName).build();
+    } catch (DuplicateKeyException ex) {
+      Optional<TestCaseLock> existingLock = testCaseLockRepository.findByTestCaseId(testCaseId);
+      if (existingLock.isPresent()) {
+        lockInfo =
+            LockInfo.builder()
+                .lockedId(testCaseId)
+                .isLocked(true)
+                .lockedBy(existingLock.get().getLockedBy())
+                .build();
```

Comment:
this will always be suppressed by line 60.

Reply:
@adongare it shouldn't be there, I removed it.